### PR TITLE
Overhaul of the GUI

### PR DESCRIPTION
**Pull request by JonasKaad**
In this PR the JFoenix library has been added to the project. 
The UI has been overhauled with JFoenix components, giving it a fresh new look. 🎇
CSS files have been added to support styling the different scenes. This also includes a possiblity to add fonts.
There has also been some minor fixes to a few different classes.
Updated the names of a few methods and variables as requested by @AlexanderNorup 
Added comments on a few core methods, such as the **_changeView_** in the **PresentationSingleton.java**

### CODE REVIEW

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/CreateCreditsController.java [hunk 1]
```diff
@@ -65,46 +82,162 @@ public void addCredits() {
...
+    public Date datePicker() {
+        LocalDate localDate = birthdayPicker.getValue();
+        Instant instant = Instant.from(localDate.atStartOfDay(ZoneId.systemDefault()));
+        Date date = Date.from(instant);
+        return date;
+    }
+
     public void seeCredits() {
+        creditsListView.setVisible(!creditsListView.isVisible());
+        if (creditsListView.isVisible()) {
```
Comment (IsabellaNorup) [inline src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/CreateCreditsController.java:146]: Bruger du visibility property'en til at tjekke om man er logget ind her?

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/FrontPageController.java [hunk 2]
```diff
@@ -188,10 +237,32 @@ public void editHandler() {
...
+    public void setupUI(){
+        fulcrum.getPrimaryStage().getScene().getStylesheets().setAll(String.valueOf(getClass().getResource("/css/style.css")));
+    }
+
+    /**
+     * TODO
+     * Method to enable darkMode. Still very basic and only for frontpage
+     */
+    public void darkMode(){
+        darkModeCounter++;
```
Comment (IsabellaNorup) [inline src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/FrontPageController.java:256]: Hvorfor er det en counter?

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/JavaFXTest.java [hunk 3]
```diff
@@ -13,13 +13,16 @@
 
     @Override
     public void start(Stage primaryStage) throws Exception {
```
Comment (IsabellaNorup) [inline src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/JavaFXTest.java:15]: Todo: Mind mig lige om at skifte navnet på den her klasse.

Det er ikke noget der skal fikses nu, men hvis det kunne få et mere meningsfyldt navn, så vil det gøre mig glad.

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/PresentationSingleton.java [hunk 4]
```diff
@@ -61,33 +58,41 @@ public String getName(){
...
         this.changeView(view, new Object());
     }
 
-    public void changeView(String view, Object addtionalData){
+    /**
+     * Method change between scenes. Two variations, one where an object is parsed and one without.
+     * @param view is the String name of the FXML document for the next scene. It gets parsed through an FXMLLoader.
+     * @param additionalData is an optional paramter, which enables the possibility of sending objects between
+     *                       scenes, such as IProgrammes or ICredits.
+     */
```
Comment (IsabellaNorup) [inline src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/PresentationSingleton.java]: Documentation is nice. Thanks 🎉

**Review by IsabellaNorup — APPROVED** [review]
Code looks fine.
I've posted some comments on a few things. Mostly just questions.

If we need any changes, we can do them after the presentation.
I want to check out the branch and actually run the code so see the changes. But I'll do that sometime Monday. Maybe we'll just take a quick look there together, so it's all ready for presentation.

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/CreateCreditsController.java (hunk 1 above)
Comment (JonasKaad) [reply src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/CreateCreditsController.java:146]: Nej, den her er til for basically at give knappen "Se Credits" mulighed for både at åbne og lukke listviewet. Den er basically ikke andet end en "on"/"off" switch. 
Ideelt set kan man kun nå til denne side hvis man allerede er logget ind og vælger et program man som producer ejer, så tænker ikke det er en nødvendighed her.

File: src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/FrontPageController.java (hunk 2 above)
Comment (JonasKaad) [reply src/main/java/dk/sdu/seb05/semesterprojekt/PresentationLayer/FrontPageController.java:256]: Det er en counter fordi... det var det første der lige faldte mig ind da jeg rodede med det første gang, helt klart noget der kan ændres og nok også skal, når det skal være muligt over flere forskellige scenes.

**Pull request merged**